src/ripple/overlay/impl/ProtocolMessage.h

```diff
@@ -120,51 +120,94 @@ buffersBegin(BufferSequence const& bufs)
         bufs);
 }
 
+/** Parse a message header
+ * @return a seated optional if the message header was successfully
+ *         parsed. An unseated optional otherwise, in which case
+ *         @param ec contains more information:
+ *         - set to `errc::success` if not enough bytes were present
+ *         - set to `errc::no_message` if a valid header was not present
+ */
 template <class BufferSequence>
 boost::optional<MessageHeader>
```

Comment:
Can we change this to `std::optional`? I think we are trying to migrate away from `boost::optional`

Reply:
We can. Shouldn't be a problem.